packages/ui/src/components/tabs/CustomizeTab.tsx

```diff
@@ -286,12 +287,30 @@ const ModuleGroup: React.FC<{
     onChange,
 }) => {
     const groupName = group || 'Default Group'
-    const groupDescription = module.groups.find(
-        (g) => g.name === groupName
-    )?.description
-    const groupExpanded = module.groups.find(
-        (g) => g.name === groupName
-    )?.expanded
+    const groupConfig = module.groups.find((g) => g.name === groupName)
+    const groupDescription = groupConfig?.description
+    const groupExpanded = groupConfig?.expanded
+    const groupIconConfig = groupConfig?.icon
+    const [groupIcon, setGroupIcon] = useState<HTMLImageElement | undefined>(
+        undefined
+    )
+    useEffect(() => {
+        switch (groupIconConfig?.type) {
+            case 'itemId':
+                getIcon(groupIconConfig.itemId, setGroupIcon)
+                break
+            case 'sprite':
+                getSprite(
+                    groupIconConfig.spriteId ?? 0,
+                    groupIconConfig.spriteIndex ?? 0,
+                    setGroupIcon
+                )
+                break
+            default:
+                break
+        }
+    }, [setGroupIcon, groupIconConfig])
```

Comment:
`useEffect` probably isn't needed here. Generally you want to avoid these as often lead to performance issues

I think you can move this to some function outside of this component and just call it here

```suggestion
// getGroupIcon function defined outside component
const groupIcon = getGroupIcon(groupIconConfig)
```